### PR TITLE
Updated Customer, Binary Search, Makefile

### PR DESCRIPTION
**Pull request by CurseCuff**
- Customer Class functions
- Binary Search university with rank number
- Edited Makefile to include Customer.o

### CODE REVIEW

File: include/Customer.h [hunk 1]
```diff
@@ -4,21 +4,99 @@
...
+
+//=====================
+//For logged in users
+//=====================
+
+	// Login and logout
+	
+	void logout();
+
+	// save university
```
Comment (T3sting123456) [inline include/Customer.h:32]: consider to remove commented codes.

File: include/Customer.h [hunk 2]
```diff
@@ -4,21 +4,99 @@
...
 #include <string>
 #include <ctime>
+#include "University.h"
 
 struct UserAccount
 {
+	int customerIndex;
 	std::string username;
 	std::string password;
+	std::string email;
```
Comment (T3sting123456) [inline include/Customer.h:14]: consider to have `using namespace std` to avoid repeating prefix

File: src/Customer.cpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
+#include <iostream>
+#include <iomanip>
+#include "../include/Customer.h"
+
+using namespace std;
+
+void printCustomerHeader();
+void printCustomer(UserAccount *curr);
+
+Customer::Customer()
```
Comment (T3sting123456) [inline src/Customer.cpp:10]: consider to implement builder pattern [https://www.geeksforgeeks.org/builder-design-pattern/](url)

File: include/Customer.h [hunk 4]
```diff
@@ -4,21 +4,99 @@
...
+	// feedback
+	// void sendFeedback(Feedback feedback);
+	void viewFeedback();
+
+	
 };
+
 class Customer
 {
 public:
```
Comment (T3sting123456) [inline include/Customer.h:67]: apply proper indendation

File: src/Customer.cpp [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    UserAccount *current = head;
+    while (current != nullptr)
+    {
+        UserAccount *temp = current;
+        current = current->next;
+        delete temp;
+    }
+}
+
+void Customer::addCustomerAtStart(std::string username, std::string password, std::string email, std::string contact, std::string dateOfBirth)
```
Comment (T3sting123456) [inline src/Customer.cpp:30]: would suggest to follow this style for better readability
```
Customer::adCustomerAtStart(
  std::string username,
  std::string password,
  std::string email, 
  std::string contact, 
  std::string dateOfBirth
)
```

**Comment by EuJin03** [comment]
Closed. Customer class has been implemented with hashmap instead.

File: include/Customer.h (hunk 1 above)
Comment (BryanHJH) [reply include/Customer.h:32]: who are you?

**Pull request closed without merging**